### PR TITLE
Remove source alias

### PR DESCRIPTION
**Pull request by davids5**
Per today's  dev call and @bkueng https://github.com/PX4/Firmware/pull/15139#discussion_r487719343

Use source for . and a prefix of ${R} set to / for nuttx and ${pwd} for linux

### CODE REVIEW

**Comment by davids5** [comment]
ehh 

Now we get 
```
etc/init.d-posix/rcS: 81: /home/david_s5/src/PX4/repos/mainline/Firmware/build/px4_sitl_default/tmp/rootfs/etc/init.d/rc.vehicle_setup: source: not found
etc/init.d-posix/rcS: 84: /home/david_s5/src/PX4/repos/mainline/Firmware/build/px4_sitl_default/tmp/rootfs/etc/init.d/rc.vehicle_setup: source: not found
```
NuttX is respecting `.` or `source`. 

From what I can tell googling "In the POSIX standard, which /bin/sh is supposed to respect, the command is . (a single dot), not source"

Any Ideas?

File: platforms/posix/src/px4/common/px4-alias.sh_in [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 # File is auto-generated by cmake compilation, do not edit.
 
+R=${pwd}
```
Comment (bkueng) [inline platforms/posix/src/px4/common/px4-alias.sh_in:5]:
```suggestion
R="$(pwd)/"
```
And replace all `sh x` with `. ${R}x`. Then it should work.

---

Alternatively we could do:
```
sh_exec() {
	. $(pwd)/$1
}
sh=sh_exec

# then use it as:
${sh} etc/init.d/rc.mc_defaults
```
And on Nuttx `set sh .` if that works.

File: ROMFS/cannode/init.d/rcS [hunk 2]
```diff
@@ -104,3 +104,4 @@ then
 fi
 
 uavcannode start
+unset R
```
Comment (MaEtUgR) [inline ROMFS/cannode/init.d/rcS:107]:
```suggestion
unset R

```

**Review by MaEtUgR — COMMENTED** [review]
Thanks for looking into this @davids5 ! I quickly checked and SITL works as expected. I can quickly check if Beat's alternative would work as well.

**Comment by MaEtUgR** [comment]
That works as well in SITL: https://github.com/PX4/Firmware/compare/pr-source-alternative-test

> And on Nuttx set sh . if that works.

Not sure where to adjust it for NuttX

File: ROMFS/cannode/init.d/rcS (hunk 2 above)
Comment (davids5) [reply ROMFS/cannode/init.d/rcS]: fixed

File: platforms/posix/src/px4/common/px4-alias.sh_in (hunk 1 above)
Comment (davids5) [reply platforms/posix/src/px4/common/px4-alias.sh_in]: This did not work. However R="`pwd`/" does.

**Comment by davids5** [comment]
@MaEtUgR @bkueng - thank you for testing and the suggestions.  

> That works as well in SITL: https://github.com/PX4/Firmware/compare/pr-source-alternative-test
> 
> > And on Nuttx `set sh .` if that works.
> 
> Not sure where to adjust it for NuttX

Given the the POSIX comments, I have gone back to using '.' everywhere with 

NuttX: R=/
Linux R="\`pwd\`/"

I am in favor of this because it shows the correct intent: "sourc-ing" ans "exec-ing" are not the same.

`sh` path_to_script - launch a process running `path_to_script` which **can not** change parent environment.
`.`  path_to_script and `source` path_to_script include the path_to_script which **can** change parent environment.

Let's  see what breaks now :)

We should be able to dump the `sh` alias as well. But I will wait for your testing

**Comment by MaEtUgR** [comment]
I did another test. Works great on arch
image

**Comment by MaEtUgR** [comment]
> We should be able to dump the sh alias as well. But I will wait for your testing

I also tried removing it in the SITL aliases and didn't see any error.

<details><summary>Output</summary>
<p>

```
[0/1] Re-running CMake...
-- PX4 version: v1.11.0-rc3-488-g891c6e24b1
-- PX4 config file: /home/maetugr/Firmware/boards/px4/sitl/default.cmake
-- PX4 config: px4_sitl_default
-- PX4 platform: posix
-- PX4 lockstep: enabled
-- cmake build type: RelWithDebInfo
-- Building for code coverage
-- ccache enabled (export CCACHE_DISABLE=1 to disable)
-- build type is RelWithDebInfo
-- PX4 ECL: Very lightweight Estimation & Control Library v1.9.0-rc1-456-gf666ebb
-- ROMFS: px4fmu_common
-- Configuring done
-- Generating done
-- Build files have been written to: /home/maetugr/Firmware/build/px4_sitl_default
[0/1] cd /home/maetugr/Firmware/build/px4_sitl_default/tmp && /home/maetugr/Firmware/Tools/sitl_run.sh /home/maetugr/Firmware/build/px4_sitl_default/bin/px4 none jmavsim none none /home/maetugr/Firmware /home/maetugr/Firmware/build/px4_sitl_default
SITL ARGS
sitl_bin: /home/maetugr/Firmware/build/px4_sitl_default/bin/px4
debugger: none
program: jmavsim
model: none
world: none
src_path: /home/maetugr/Firmware
build_path: /home/maetugr/Firmware/build/px4_sitl_default
empty model, setting iris as default
SITL COMMAND: "/home/maetugr/Firmware/build/px4_sitl_default/bin/px4" "/home/maetugr/Firmware/build/px4_sitl_default"/etc -s etc/init.d-posix/rcS -t "/home/maetugr/Firmware"/test_data
INFO  [px4] Creating symlink /home/maetugr/Firmware/build/px4_sitl_default/etc -> /home/maetugr/Firmware/build/px4_sitl_default/tmp/rootfs/etc

______  __   __    ___ 
| ___ \ \ \ / /   /   |
| |_/ /  \ V /   / /| |
|  __/   /   \  / /_| |
| |     / /^\ \ \___  |
\_|     \/   \/     |_/

px4 starting.

INFO  [px4] Calling startup script: /bin/sh etc/init.d-posix/rcS 0
Info: found model autostart file as SYS_AUTOSTART=10016
INFO  [param] selected parameter default file eeprom/parameters_10016
[param] Loaded: eeprom/parameters_10016
INFO  [dataman] Unknown restart, data manager file './dataman' size is 11798680 bytes
INFO  [simulator] Waiting for simulator to accept connection on TCP port 4560
Buildfile: /home/maetugr/Firmware/Tools/jMAVSim/build.xml

make_dirs:

compile:

create_run_jar:

copy_res:

BUILD SUCCESSFUL
Total time: 0 seconds
Options parsed, starting Sim.
Starting GUI...
INFO  [simulator] Simulator connected on TCP port 4560.
Init MAVLink
INFO  [commander] LED: open /dev/led0 failed (22)
INFO  [init] Mixer: etc/mixers/quad_w.main.mix on /dev/pwm_output0
INFO  [mavlink] mode: Normal, data rate: 4000000 B/s on udp port 18570 remote port 14550
INFO  [mavlink] mode: Onboard, data rate: 4000000 B/s on udp port 14580 remote port 14540
INFO  [mavlink] mode: Onboard, data rate: 4000 B/s on udp port 14280 remote port 14030
INFO  [logger] logger started (mode=all)
INFO  [logger] Start file log (type: full)
INFO  [logger] [logger] ./log/2020-10-15/14_08_09.ulg
INFO  [logger] Opened full log file: ./log/2020-10-15/14_08_09.ulg
INFO  [mavlink] MAVLink only on localhost (set param MAV_BROADCAST = 1 to enable network)
INFO  [px4] Startup script returned successfully
pxh> INFO  [mavlink] partner IP: 127.0.0.1
INFO  [ecl/EKF] 2564000: reset position to last known position
INFO  [ecl/EKF] 2564000: reset velocity to zero
INFO  [ecl/EKF] 8664000: EKF aligned, (baro hgt, IMU buf: 18, OBS buf: 14)
INFO  [ecl/EKF] 9884000: GPS checks passed

pxh> 
pxh> 
pxh> commINFO  [ecl/EKF] 14416000: reset position to GPS
INFO  [ecl/EKF] 14416000: reset velocity to GPS
INFO  [ecl/EKF] 14416000: starting GPS fusion
pxh> commanderINFO  [tone_alarm] home set
ERROR [tone_alarm] notify negative
pxh> commander takeoff
pxh> INFO  [commander] Armed by internal command
INFO  [navigator] Using minimum takeoff altitude: 2.50 m
INFO  [commander] Takeoff detected

PX4 Exiting...
pxh> Exiting NOW.
ninja: build stopped: interrupted by user.
make: *** [Makefile:226: px4_sitl] Interrupt
```

</p>
</details>

**Comment by davids5** [comment]
> > We should be able to dump the sh alias as well. But I will wait for your testing
> 
> I also tried removing it in the SITL aliases and didn't see any err

@MaEtUgR - OK removed the sh alias as well. 

@dagar once this passes CI it should be ready to merge.

**Review by dagar — APPROVED** [review]

**Comment by MaEtUgR** [comment]
SITL works as expected now on Arch and Cygwin. Thanks! NuttX fix on Cygwin still pending. I'll have another look at it.

**Pull request merged**